pkgs/development/interpreters/python/default.nix

```diff
@@ -21,7 +21,7 @@ with pkgs;
     , pythonOnTargetForTarget
     , self # is pythonOnHostForTarget
     }: let
-      pythonPackages = callPackage
+      pythonPackagesFan = callPackage
```

Comment:
Rename is not technically necessary if it's a rec set. If it's for clarity: What does `Fan` stand for out of curiosity?

Reply:
I had quite some trouble getting this to work and in the end this just happened to work without changing to much of the existing code. I can try rec and see if that works better. Fan is just another form of Fun because I needed another variable.

Reply:
Okay, I think there's a cleaner solution for what you are trying to do. The following patch should apply on your branch:

```patch
diff --git a/pkgs/development/interpreters/python/default.nix b/pkgs/development/interpreters/python/default.nix
index fea5a992602..551697d2f05 100644
--- a/pkgs/development/interpreters/python/default.nix
+++ b/pkgs/development/interpreters/python/default.nix
@@ -21,7 +21,7 @@ with pkgs;
     , pythonOnTargetForTarget
     , self # is pythonOnHostForTarget
     }: let
-      pythonPackagesFan = callPackage
+      pythonPackages = callPackage
         ({ pkgs, stdenv, python, overrides }: let
           pythonPackagesFun = import ../../../top-level/python-packages.nix {
             inherit stdenv pkgs lib;
@@ -69,17 +69,17 @@ with pkgs;
               recursivePthLoader
             ;
           };
+          aliases = self: super: lib.optionalAttrs (config.allowAliases or true)
+            (import ../../../top-level/python-aliases.nix lib super);
         in lib.makeScopeWithSplicing
           pkgs.splicePackages
           pkgs.newScope
           otherSplices
           keep
-          (lib.extends overrides pythonPackagesFun))
+          (lib.extends (lib.composeExtensions aliases overrides) pythonPackagesFun))
         {
           overrides = packageOverrides;
         };
-        aliases = lib.optionalAttrs (config.allowAliases or true) (import ../../../top-level/python-aliases.nix lib pythonPackagesFan);
-        pythonPackages = pythonPackagesFan // aliases;
     in rec {
         isPy27 = pythonVersion == "2.7";
         isPy35 = pythonVersion == "3.5";
```

Side note: a `rec` set would definitely helped in now place here, I think.